### ISSUE TITLE
Add event categories, with a color for each.

### ISSUE THREAD

**Issue by smenjas** [smenjas/calendar-vanilla#14]
This feature depends on #12.

I'd like to be able to categorize events, and assign colors to categories.

It would be nice to have some categories defined by default. For example:
- Exercise, green
- Study, red
- Music, blue
- Reading, yellow
- Fun, orange
- Chores, purple
- Work, black

There are issues with editing categories if they have events assigned to them already:
- If we edit a category with events, we should ask whether to unassign those events.
- If we change a category color to match another color, should we warn about that?

Other considerations:
- Let's limit which colors are available for now. It's just too easy to pick unreadable color combos.
- Support both light and dark modes, with dark versions of colors for light mode, and light versions of colors for dark mode.
- Allow black/white as a color, with black in light mode and white in dark mode to represent the same category? (Let's call this "inverse".)
- Should we allow white in light mode and black in dark mode in addition to the "inverse" scheme described above? (Let's call this "colorless".)
- See #13.
